File: ml/ml_service.py
from flask import Flask, request, jsonify
from model_utils import recommend_recipe_precise, get_recipe_steps

app = Flask(__name__)

@app.route("/recommend", methods=["POST"])
def recommend():
    data = request.json
    ingredients = data.get("ingredients", "").strip()

    recommendations = recommend_recipe_precise(ingredients)

    if recommendations.empty:
        return jsonify([])

    return jsonify(recommendations.to_dict(orient="records"))


@app.route("/recipe/<name>", methods=["GET"])
def recipe(name):
    recipe = get_recipe_steps(name)

    if not recipe:
        return jsonify({"error": "Recipe not found"}), 404

    return jsonify(recipe)


if __name__ == "__main__":
    app.run(port=5001, debug=True)

File: backend/services/ml.service.js
import axios from "axios";

export const getRecommendations = async (ingredients) => {
    const response = await axios.post(
        "http://localhost:5001/recommend",
        { ingredients }
    );

    return response.data;
}

export const getRecipe = async (name) => {
    const response = await axios.get(
        `http://localhost:5001/recipe/${name}`
    );

    return response.data;
}

